### ISSUE TITLE
Tuya/Jinvoo/Heiman curtain switch (SM-SW101-ZC)

### ISSUE THREAD

**Issue by Pedder007** [dresden-elektronik/deconz-rest-plugin#2999]
## Device

- Product name: SM-SW101-ZC
- Manufacturer: Tuya/Jinvoo (Heiman, read from the node list as vendor)
- Model identifier: TS0302
- Device type: 
  - [ ] Light
  - [ ] Lock
  - [ ] Remote
  - [ ] Sensor
  - [ ] Siren
  - [ ] Thermostat
  - [X] Other: Smart Wall Switch / Curtain Switch

## Screenshots

![BasicClusterInfo](https://user-images.githubusercontent.com/44653371/86176249-f7de9e00-bb24-11ea-8e8d-c4c7c2a4b537.JPG)

![NodeInfo01](https://user-images.githubusercontent.com/44653371/86176295-0331c980-bb25-11ea-8132-1a5c7d75fad4.JPG)

![NodeInfo02](https://user-images.githubusercontent.com/44653371/86176351-180e5d00-bb25-11ea-9ead-6e82f97f498b.JPG)

![Config01](https://user-images.githubusercontent.com/44653371/86176380-22305b80-bb25-11ea-8cac-298de3fe5529.JPG)
![Config02](https://user-images.githubusercontent.com/44653371/86176391-25c3e280-bb25-11ea-9f17-23d27d7d0afd.JPG)

![ClusterInfo Window Covering01](https://user-images.githubusercontent.com/44653371/86176410-2c525a00-bb25-11ea-916e-7d027722f44a.JPG)
![ClusterInfo Window Covering02](https://user-images.githubusercontent.com/44653371/86176416-2eb4b400-bb25-11ea-810c-7c47e0e2669a.JPG)
![ClusterInfo Window Covering03](https://user-images.githubusercontent.com/44653371/86176426-307e7780-bb25-11ea-8887-e7710ca6617b.JPG)
![ClusterInfo Window Covering04](https://user-images.githubusercontent.com/44653371/86176432-32e0d180-bb25-11ea-823b-aa5de1de98b3.JPG)
![ClusterInfo Window Covering05](https://user-images.githubusercontent.com/44653371/86176436-34aa9500-bb25-11ea-8dfb-e7d7a61db362.JPG)

**Comment by Pedder007** [comment]
Update Basic Cluster Info (did reading again)
![BasicClusterInfo n](https://user-images.githubusercontent.com/44653371/86387006-d3a0cf80-bc92-11ea-98e4-02c68c4f79ab.JPG)

Having now the manufacturer name in the field: _TYZB01_dazsid15
I searched an found this: https://github.com/Koenkk/zigbee-herdsman-converters/issues/621

I Also tested to steer the device direct from the deConz Hui and it worked:
![ClusterInfo Window Covering](https://user-images.githubusercontent.com/44653371/86387201-18c50180-bc93-11ea-82c0-43e101504f3a.JPG)

So it seems already to work with the ConBee II stick and deConz, but not in Pimatic, which might have to be solved there then

**Comment by SwoopX** [comment]
Closing this as device is now supported with .79 beta.

**Comment by Pedder007** [comment]
Hey @SwoopX, thanks for your effort 👍
... but I‘m a bit confused 😊What does ‚supported with .79 beta‘ mean?
Can I use the device now by Pimatic?

**Comment by SwoopX** [comment]
When you have subscribed to the apt beta channel and updated to the current version, then yes. Also marthoc should have updated the container already.

**Comment by Pedder007** [comment]
Not quiet sure what you mean, but I will spend some research and convinced that I will find the answer 😉
Thanks again!

**Comment by Pedder007** [comment]
Hi again @SwoopX, I'm sorry but I couldn't find anything which seems to fit to your hint. I still have no clue at all what I have to update from where. I'm running Pimatic with the raspbee 0.1.5 plugin on a Raspi4 with a ConBee II and deConz in headless mode. 
Phoscon Vers. is 2.05.75 and stick firmware is 264A0700. 
Could you support with a link? That would be really great :-)

**Comment by SwoopX** [comment]
@Pedder007 I'm afraid I dont know Pimatic so I probably cannot help you there. How have you installed deconz anyway. You are already 4 versions behind 😉

If it is a package pimatics folks provide, they maybe should update.

**Comment by Pedder007** [comment]
Ok, I understand this an searched a bit more. I think I now also got your point. You updated/enhanced the latest deconz version 2.05.79... to use the curtain switch, right?
I was a bit lost as you only wrote .79 :-)
If this is the case, then it‘s definitely a question of integration in the respective Pimatic plugin.
As written somewhere, via the Deconz-gui I have already been able to steer the device, also with .75.
Thanks for now then 👍

**Comment by rkupfer2020** [comment]
Hi, 
I use the same Heimann / _TYZB01_dazsid15 curtain switch with Conbee and Deconz. I already upated the Docker Container to 2.05.79. But I am still not able to use the Deconz GUI to open/close the curtains. Only with VNC they respond to the commands.
Under Home Assistant I can open and stop the curtains, but not close them. Anybody has an idea how to solve the issue?
Best regards
Roger

**Comment by SwoopX** [comment]
> But I am still not able to use the Deconz GUI to open/close the curtains.

I assume you mean Phoscon in your case. This is kind of normal as devices have to be added seperately. I'd recommend to raise a corresponding request at the Phoscon repository.

> Only with VNC they respond to the commands.

This is actually the deconz GUI. If all works well there, then the device doesn't have any issue.

> Under Home Assistant I can open and stop the curtains, but not close them.

Probably the wrong API commands are used in this case. See https://github.com/dresden-elektronik/deconz-rest-plugin/issues/3063

**Comment by cobirnm** [comment]
> ## Device
> * Product name: SM-SW101-ZC
> * Manufacturer: Tuya/Jinvoo (Heiman, read from the node list as vendor)
> * Model identifier: TS0302
> * Device type:
>   
>   * [ ]  Light
>   * [ ]  Lock
>   * [ ]  Remote
>   * [ ]  Sensor
>   * [ ]  Siren
>   * [ ]  Thermostat
>   * [x]  Other: Smart Wall Switch / Curtain Switch
> 
> ## Screenshots
> ![BasicClusterInfo](https://user-images.githubusercontent.com/44653371/86176249-f7de9e00-bb24-11ea-8e8d-c4c7c2a4b537.JPG)
> 
> ![NodeInfo01](https://user-images.githubusercontent.com/44653371/86176295-0331c980-bb25-11ea-8132-1a5c7d75fad4.JPG)
> 
> ![NodeInfo02](https://user-images.githubusercontent.com/44653371/86176351-180e5d00-bb25-11ea-9ead-6e82f97f498b.JPG)
> 
> ![Config01](https://user-images.githubusercontent.com/44653371/86176380-22305b80-bb25-11ea-8cac-298de3fe5529.JPG)
> ![Config02](https://user-images.githubusercontent.com/44653371/86176391-25c3e280-bb25-11ea-9f17-23d27d7d0afd.JPG)
> 
> ![ClusterInfo Window Covering01](https://user-images.githubusercontent.com/44653371/86176410-2c525a00-bb25-11ea-916e-7d027722f44a.JPG)
> ![ClusterInfo Window Covering02](https://user-images.githubusercontent.com/44653371/86176416-2eb4b400-bb25-11ea-810c-7c47e0e2669a.JPG)
> ![ClusterInfo Window Covering03](https://user-images.githubusercontent.com/44653371/86176426-307e7780-bb25-11ea-8887-e7710ca6617b.JPG)
> ![ClusterInfo Window Covering04](https://user-images.githubusercontent.com/44653371/86176432-32e0d180-bb25-11ea-823b-aa5de1de98b3.JPG)
> ![ClusterInfo Window Covering05](https://user-images.githubusercontent.com/44653371/86176436-34aa9500-bb25-11ea-8dfb-e7d7a61db362.JPG)

Have you been able to Calibrate the Switch?

**Comment by djfanatix** [comment]
Did you try to modify 0x0017 or 0x007 to 0x002 (calibration mode)? Does this work?

**Comment by Pedder007** [comment]
Hi, sorry didn't find the time to continue here earlier.
I now just tested again (after first updating the PhosconApp and ConBee II firmware to newest releases).
The PhosconApp now shows the device, but as a 'light'!? This happened three times now, allthough I used the discovering mode for switches.
Via the deconz-gui I was able to switch up/down/close, so for me nothing seems having changed to the status above.
I fear as long as the device is only discovered as a light I will also not get it integrated in my HO Pimatic.
